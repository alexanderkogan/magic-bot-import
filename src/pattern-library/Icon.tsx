import * as React from 'react'

// see: https://pattern-library.portal.otto.market/fragments#icons
export type IconType =
    | 'arrowleft'
    | 'arrowright-bold'
    | 'calendar'
    | 'check-bold'
    | 'info'
    | 'menu'
    | 'minus-bold'
    | 'plus-bold'
    | 'search'
    | 'upload'
    | 'trash'
    | 'close'
    | 'settings'
    | 'circle'
    | 'help'
    | 'success'
    | 'arrowdown'
    | 'arrowleft-bold'
    | 'arrowup'
    | 'error'
    | 'viewtoggle-list'
    | 'tooltip'
    | 'close-bold'
    | 'plus'
    | 'edit'
    | 'check'
    | 'warning'
    | 'arrowdown-bold'
    | 'arrowright'
    | 'arrowup-bold'
    | 'download'
    | 'viewtoggle-tiles'

export interface IconProps {
    type: IconType
}

export default (props: IconProps) => <i className={`obc_icon-${props.type}`}></i>
