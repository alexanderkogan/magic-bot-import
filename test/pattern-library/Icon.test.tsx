import * as React from 'react'

import Icon from '../../src/pattern-library/Icon'
import { mount } from '../test-utils'

describe('Icon', () => {
    it('renders', () => {
        const element = mount(<Icon type="arrowdown-bold" />)
        expect(element).toMatchSnapshot()
    })
})
