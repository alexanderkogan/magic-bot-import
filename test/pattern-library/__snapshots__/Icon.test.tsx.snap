// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders 1`] = `
<Component
  type="arrowdown-bold"
>
  <i
    className="obc_icon-arrowdown-bold"
  />
</Component>
`;
